util/testutil/protobuf.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2023 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package testutil
+
+import (
+	"bytes"
+	"encoding/binary"
+
+	"github.com/gogo/protobuf/proto"
+	dto "github.com/prometheus/client_model/go"
```

Comment:
As above, this has to be dto "github.com/prometheus/prometheus/prompb/io/prometheus/client" . Also, style nit, please separate the named imports by a blank line.

Reply:
And only now I realize how this happened: You are acting on protobufs that come from client_golang (which doesn't use gogo-protobuf, but the official variant). 

I know how annoying that is, but we should keep each version to itself. So how about just leaving the code in `model/textparse/protobufparse_test.go` alone (with its separate implementation of encoding the protobuf) and move this file into the scrape package, calling out that it acts on the client_golang protobufs.

Reply:
I can do that, I've contributed the test. I thought it's ok to change protobufparse_test.go since it needs to be able to parse what client_golang sends anyway.

Reply:
Please check https://github.com/zenador/prometheus/pull/2

Reply:
Yeah, so the protobuf binary format created by either package is the same. But the Go types aren't the same.

Reply:
@krajorama Thank you!